app/editor.js

```diff
@@ -445,7 +447,20 @@ ved.cql.renderItems = function(sel, group, indexPrefix) {
         mode: 'vega-lite',
         actions: {export: false}
       };
-      vg.embed(id, opt);
+      vg.embed(id, opt, function(err, result) {
+        if(err) {
+          console.error("[CompassQL]", err);
+          return;
+        }
+
+        // unregister signals for unused views
+        if (ved.cql.views.get(id)) {
```

Comment:
One thing that you might wanna do besides removing things in exit() is moving `exit()` to run before `enter()` so removed views already get removed from the dictionary before enter is running here.